### ISSUE TITLE
Write access for the files opened from other applications

### ISSUE THREAD

**Issue by maxistar** [maxistar/TextPad#86]
when file is being opened from other app e.g. from file browser it is being opened as read only and therefore it is not possible to store it. We need to fix the issue of instruct used how to proceed.

**Comment by maxistar** [comment]
Here is the another user mentioning the same problem:
Prerequisites

- Will be written “STE” = Simple Text Editor.

- The settings are the default ones, and the encoding is in UTF-8 as set by default and which is universal.

- I also tried to change the encoding to other generic formats like ISO-8859-1 but even for this other type of encoding your application behaves exactly the same way.  And so the same problems.

          I installed your appli. on my smartphone  STE under version 1.24.1

It appears from the data on the Play Store to have been updated recently, on July 31, 2023.

I found it at first glance ; very simple to use, includes everything you need to quickly write notes, information, memos, etc.

I'm seeing issues with its use.

1) After creating a text.  And after saving, STE does not know how to re-open the text that it has just saved ?

Before leaving the appli. if I forget to save the STE file, ask the question should it be saved, which is an exact operation.

2) After exiting the app.  And restart it
for the CMD: open a recent file,

2a) The latest backup file does not appear in the list.

2b) Opening a file via the CMD "recent file" and choosing 1 file from the list offered (6 files are displayed) never works.
The response is "unable to read file"

3) After opening the appli.  STE, opening a ×××.txt file ( the same I previously saved ) does not work.

4) If the ×××.txt file is written with another appli. STE knows how to open it.

5) The latter (file) cfr.  (4) saved by STE, the appli.  STE no longer knows how to re-open it.  cfr.  (1)

I have to pass the file through another application which after saving by this appli. STE then knows how to open the file modified by this app.

6) Opening a file.  ×××.Txt by first choosing the file to open with an explorer "fichiers" from Samsung or "files" from Google,  Cx Explorateur appears on the smartphone screen all appli.  likely to be able to open a .txt file (see screenshot)
The app.  STE knows how to open the file.  txt.

7) When STE starts, a blank page is displayed with the text “Type a text here”
In this case, request the opening of a ×××.Txt file with the CMD “open” without saving the current file the appli.  STE does nothing, stays on the current file and the information "unable to open the file".


In summary,

I) STE (Simple Text Editor) knows how to write a ×××.txt file and save it.
He doesn't know how to re-open the file he just saved !

II) Open a ×××.txt file with a file explorer (the file explorer displays the different appli. that are likely to be able to edit the file) as well as the appli.  text editors included in internet browsers like;  Brave, Chrome, HTML Viewer, etc... then only in this case " STE " is able to edit the file.  txt: cfr.  screenshot, (see the yellow tags)

Have you received feedback from other users of your STE software who have reported these errors to you ?

Looking forward to reading you and providing me with answers to my questions.

Best regards

**Comment by maxistar** [comment]
I still can not find a solution for this, the best I can do at the moment to instruct user that file is opened in read only mode and ask to open it again using open menu of seva with other name...

**Comment by David-DeZ** [comment]
This issue is also addressed in #95 - where I just wrote:        I have the same issue - Android 14 Samsung A52 - as I wrote in my Google Play review - "Great app - does a basic fundamental function that for some reason is otherwise lacking in Android. BUT although works flawlessly from the app (open, edit, save, save as, etc) it does not from a file manager, there it opens and edits but does not overwrite when choosing save. Please fix and I'll up to the 5* it otherwise deserves."

**Comment by ludwigseibt** [comment]
I love this app for its simplicity, but this bug is keeping me from using it. I have a notes.txt file on my Dropbox. When I open it from Dropbox in SimpleTextEditor, a conflicting copy gets created on save. Same thing happens, when SimpleTextEditor opens that notes file as last opened file.

**Comment by David-DeZ** [comment]
The app now shows a notification that the file is "Read Only" when opened from a file manager and gives the option to "Open Again", which gives you a list of files, from which you can again choose the file. This puts you into the editable file, and 'Bob's your uncle" ("piece of cake" or "easy peasy"). IMHO, considering this is a SIMPLE Text Editor, it is a reasonable limitation and  the issue should be closed.
